runtime/context.go

```diff
@@ -271,3 +271,36 @@ func (c *contextLogger) Warn(ctx context.Context, msg string, userFields ...zap.
 func (c *contextLogger) Check(lvl zapcore.Level, msg string) *zapcore.CheckedEntry {
 	return c.log.Check(lvl, msg)
 }
+
+// ContextMetrics emits metrics
+type ContextMetrics struct {
+	scope                  tally.Scope
+	InboundHTTPMetrics     *InboundHTTPMetrics
+	EndpointMetrics        *EndpointMetrics
+	InboundTChannelMetrics *InboundTChannelMetrics
```

Comment:
The abstraction is awkward here, the field is irrelevant when the struct is part of the http endpoint, vise versa for the http metrics field.

Reply:
fixed